functions/v2/typed/googlechatapp/main.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# [START functions_typed_googlechatbot]
+import functions_framework
+
+
+class ChatRequest:
+    message: dict
+
+    # Required to deserialize the request
+    @staticmethod
+    def from_dict(req: dict) -> "ChatRequest":
+        obj = ChatRequest()
+        obj.message = req["message"]
+        return obj
+
+
+class ChatResponse:
```

Comment:
Feel free to ack. This https://docs.python.org/3/library/dataclasses.html is something I have used before for reducing boilerplate code which is pretty neat. Not sure if it is applicable here.

Reply:
dataclasses look great for this application, went ahead and adopted them. Thanks Kenneth!